Content.Server/ReagentOnItem/EntitySystems/ReagentOnItemSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Item;
+using Content.Shared.Popups;
+using Content.Shared.ReagentOnItem;
+using Content.Server.Fluids.EntitySystems;
+using Content.Shared.Clothing.Components;
+using Content.Shared.IdentityManagement;
+
+namespace Content.Server.ReagentOnItem;
+
+/// <summary>
+///     The system that helps deal with a genaric reagent being applied to items.
+/// </summary>
+public sealed class ReagentOnItemSystem : EntitySystem
+{
+    [Dependency] private readonly PuddleSystem _puddle = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+    /// <summary>
+    ///     The function tries to apply the given solution mixture to the item.
+    ///     It will apply all reagents that can stick and will spill anything
+    ///     that cannot stick to the floor.
+    /// </summary>
+    /// <returns> Returns false if there is no solution or if entity isn't an item and true otherwise. </returns>
+    public bool AddReagentToItem(EntityUid item, Solution reagentMixture)
+    {
+        if (reagentMixture.Volume <= 0 || !HasComp<ItemComponent>(item))
+        {
+            return false;
+        }
+
+        // Make sure the item doesn't have a non stick surface (E.g jani gloves or boots)
+        if (!HasComp<NonStickSurfaceComponent>(item))
+        {
+            // Yes this code is sussy but its much better than what was used before.
+            // All suspect code for this system is contained here (I hope).
+
+            // Remove all reagents that can actually stick to things and apply them
+            // to the item.
+            var volSpaceLube = reagentMixture.RemoveReagent("SpaceLube", reagentMixture.Volume).Double();
+            var volSpaceGlue = reagentMixture.RemoveReagent("SpaceGlue", reagentMixture.Volume).Double();
+
+            if (volSpaceLube > 0)
+            {
+                var lubed = EnsureComp<SpaceLubeOnItemComponent>(item);
+                AddAndDetermineExtra(lubed, "SpaceLube", volSpaceLube, reagentMixture);
+            }
+            if (volSpaceGlue > 0)
+            {
+                var glued = EnsureComp<SpaceGlueOnItemComponent>(item);
+                AddAndDetermineExtra(glued, "SpaceGlue", volSpaceGlue, reagentMixture);
+            }
+        }
+        else
+        {
+            // This means the item had a nonstick surface.
+            _popup.PopupEntity(Loc.GetString("non-stick-gloves-reagent-falls-off", ("target", Identity.Entity(item, EntityManager))), item);
+        }
```

Comment:
Flip this and maybe make it a guard clause if possible.

Reply:
The reason I didn't use a guard clause is because in both paths you could spill so I'd have to duplicate this line:
`_puddle.TrySpillAt(item, reagentMixture, out var puddle, false);`

Its definitely clearer as a guard statement but I was worried about doubling up on that line. What do you think?

Reply:
Doubling is fine here since it makes the flow easier to read.